test/stress/queries-performance-analyzer.js

```diff
@@ -0,0 +1,229 @@
+  AND (fiat > 0)
+  ORDER BY created DESC limit $4 offset $5`
+
+  return db.any(cashOutSql, [REDEEMABLE_AGE, from, until, limit, offset, id, txClass, machineName, customerName, fiatCode, cryptoCode, toAddress, status])
+}
+
+function getTx (txId, txClass) {
+  const cashInSql = `EXPLAIN ANALYZE
+  select 'cashIn' as tx_class, txs.*,
+  ((not txs.send_confirmed) and (txs.created <= now() - interval $1)) as expired
+  from cash_in_txs as txs
+  where txs.id=$2`
+
+  const cashOutSql = `EXPLAIN ANALYZE
+  select 'cashOut' as tx_class,
+  txs.*,
+  (extract(epoch from (now() - greatest(txs.created, txs.confirmed_at))) * 1000) >= $2 as expired
+  from cash_out_txs txs
+  where txs.id=$1`
+
+  return txClass === 'cashIn'
+    ? db.any(cashInSql, [cashInTx.PENDING_INTERVAL, txId])
+    : db.any(cashOutSql, [txId, REDEEMABLE_AGE])
+}
+
+function getTxAssociatedData (txId, txClass) {
+  const billsSql = `EXPLAIN ANALYZE select 'bills' as bills, b.* from bills b where cash_in_txs_id = $1`
+  const actionsSql = `EXPLAIN ANALYZE select 'cash_out_actions' as cash_out_actions, actions.* from cash_out_actions actions where tx_id = $1`
+
+  return txClass === 'cashIn'
+    ? db.any(billsSql, [txId])
+    : db.any(actionsSql, [txId])
+}
+
+const run = () => {
+  const deviceId = '7526924341dc4a57f02b6411a85923de' // randomly generated by the load script
+  const customerId = '99ac9999-9999-99e9-9999-9f99a9999999' // hardcoded on the current load script
+  const cashOutTxId = 'c402a7ae-b8f7-4781-8080-1e9ab76d62b5' // randomly generated by the load script
+  const cashInTxId = '4d8d89f4-7d77-4d30-87e8-be9de05deea7' // randomly generated by the load script
+
+  const getExecutionTime = _.compose(_.get('QUERY PLAN'), _.last)
+  Promise.all([filterCustomer(), filterTransaction(), getCustomerById(customerId), simpleGetMachineLogs(deviceId), batchCashIn(), batchCashOut(),
+    getTx(cashInTxId, 'cashIn'), getTx(cashOutTxId, 'cashOut'), getTxAssociatedData(cashInTxId, 'cashIn'), getTxAssociatedData(cashOutTxId, 'cashOut')])
+    .then(([filterCustomer, filterTransaction, getCustomerById, logs, batchCashIn, batchCashOut, getTxCashOut, getTxCashIn,
+      getTxAssociatedDataCashIn, getTxAssociatedDataCashOut]) => {
+      console.log(`filterCustomer => ${getExecutionTime(filterCustomer)}`)
+      console.log(`filterTransaction => ${getExecutionTime(filterTransaction)}`)
+      console.log(`getCustomerById => ${getExecutionTime(getCustomerById)}`)
+      console.log(`batchCashOut + batchCashIn => ${getExecutionTime(batchCashOut) + ' + ' + getExecutionTime(batchCashIn)}  `)
+      console.log(`getTx (cash-out) => ${getExecutionTime(getTxCashOut)}`)
+      console.log(`getTx (cash-in)  => ${getExecutionTime(getTxCashIn)}`)
+      console.log(`getTxAssociatedData (cash-in)  => ${getExecutionTime(getTxAssociatedDataCashIn)}`)
+      console.log(`getTxAssociatedDataCashOut (cash-out)  => ${getExecutionTime(getTxAssociatedDataCashOut)}`)
+    })
+}
```

Comment:
Would it be a good idea to export the results to a file or something, in case we want to preserve the results over a period of time?